build/cmake/init.cmake

```diff
@@ -36,6 +36,36 @@ if(MSVC)
               "Flags used by the CXX compiler during ${cfg_upper} builds." FORCE)
         endif()
     endforeach()
+    
+    if(wxUSE_OPTIMISE)
+        set(MSVC_LINKER_RELEASE_FLAGS " /LTCG /OPT:REF /OPT:ICF")
+        wx_string_append(CMAKE_EXE_LINKER_FLAGS_RELEASE "${MSVC_LINKER_RELEASE_FLAGS}")
+        wx_string_append(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${MSVC_LINKER_RELEASE_FLAGS}")
+        wx_string_append(CMAKE_STATIC_LINKER_FLAGS_RELEASE " /LTCG")
+        set(MSVC_COMPILER_RELEASE_FLAGS " /Ox /Oi /Ot /Oy /GS- /Gy /GL /Gw")
+        wx_string_append(CMAKE_CXX_FLAGS_RELEASE "${MSVC_COMPILER_RELEASE_FLAGS}")
+        wx_string_append(CMAKE_C_FLAGS_RELEASE "${MSVC_COMPILER_RELEASE_FLAGS}")
+    endif()
+    
+    if(NOT wxUSE_STRIPPED_RELEASE)
+        set(MSVC_PDB_FLAG " /DEBUG")
+    endif()
+    wx_string_append(CMAKE_EXE_LINKER_FLAGS_RELEASE "${MSVC_PDB_FLAG}")
+    wx_string_append(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${MSVC_PDB_FLAG}")
+elseif(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"))
+    if(wxUSE_OPTIMISE)
+        set(GCC_PREFERRED_RELEASE_FLAGS " -O2 -fomit-frame-pointer")
+        wx_string_append(CMAKE_CXX_FLAGS_RELEASE "${GCC_PREFERRED_RELEASE_FLAGS}")
+        wx_string_append(CMAKE_C_FLAGS_RELEASE "${GCC_PREFERRED_RELEASE_FLAGS}")
+    endif()
+    
+    if(wxUSE_STRIPPED_RELEASE)
+        set(LD_STRIPPING_FLAG " -s")
+    else()
+        set(LD_STRIPPING_FLAG " -g")
```

Comment:
If I look at [LD documentation](https://sourceware.org/binutils/docs/ld/Options.html): `-g Ignored. Provided for compatibility with other tools.`

I think you need to add `-g` to CXX/C flags when `NOT wxUSE_STRIPPED_RELEASE`, and add `-s` to  LD flags when `wxUSE_STRIPPED_RELEASE`.

Reply:
Thank you, you are absolutely right. I have fixed it.